shuffle/cli/src/shared.rs

```diff
@@ -269,6 +267,58 @@ impl Home {
             )),
         }
     }
+
+    pub fn write_network_into_config(&self) -> Result<()> {
+        let network_config_path = self.shuffle_path.join("Networks.toml");
+        let network_config_string = toml::to_string_pretty(&generate_localhost_network()?)?;
+        fs::write(network_config_path, network_config_string)?;
+        Ok(())
+    }
+}
+
+fn generate_localhost_network() -> Result<Value> {
+    let local_host_value = r#"
+        [localhost]
+        base = 'http://127.0.0.1'
+        json_rpc_port = 8080
+        dev_api_port = 8081
+    "#;
+    Ok(toml::from_str::<Value>(local_host_value)?)
+}
+
+pub fn build_url(network_name: &str) -> Result<Url> {
+    let home = Home::new(get_home_path().as_path())?;
+    let network_toml_contents = fs::read_to_string(home.get_networks_config_path())?;
+    let network_toml: Value = toml::from_str(network_toml_contents.as_str())?;
+    check_network_exists_in_toml(network_name, &network_toml)?;
+    let network_value = &network_toml[network_name];
+    Ok(Url::from_str(
+        format!(
+            "{}:{}",
+            parse_toml_string_value(&network_value["base"]),
+            network_value["dev_api_port"].to_string()
+        )
+        .as_str(),
+    )?)
+}
+
+fn check_network_exists_in_toml(network_name: &str, all_networks: &Value) -> Result<()> {
+    panic::set_hook(Box::new(|_info| {
+        // Allows us to return the error below instead of panic!
+    }));
+
+    let result = panic::catch_unwind(|| &all_networks[network_name]);
```

Comment:
Let's remove all these panics. We should haven't to catch a panic to handle an unknown network.

Reply:
Accessing a field in a toml that doesn't exist actually panics. I was trying to catch the panic and smoothly error out and the combination of the set_hook and catch_unwind helps me do this. Looking at the codebase there is one other instance of someone using set_hook to catch their panic. I found that specific example i used in the code online.

Reply:
<img width="1144" alt="Screen Shot 2021-11-05 at 4 13 39 PM" src="https://user-images.githubusercontent.com/55404786/140588187-a080e109-a25c-4575-85da-fd69afc1d371.png">

Reply:
Furthermore, 
1. how would you test this?
2. where are the tests?

Reply:
This is avoided if you check for the existence of the key before you index. `panic` should never be in an expected flow.

Reply:
1. I had tests that cover this case in shared.rs at the very bottom where I'd just check if that function errored.


The panic doesn't happen with my new PR.